src/Microsoft.OData.Edm/Schema/EdmNavigationProperty.cs

```diff
@@ -249,6 +260,24 @@ private static IEdmEntityType GetEntityType(IEdmTypeReference type)
 
             return entityType;
         }
+        private static IEdmComplexType GetComplexType(IEdmTypeReference type)
+        {
+            IEdmComplexType complexType = null;
+            if (type.IsComplex())
+            {
+                complexType = (IEdmComplexType)type.Definition;
+            }
+            else if (type.IsCollection())
+            {
+                type = ((IEdmCollectionType)type.Definition).ElementType;
+                if (type.IsComplex())
+                {
+                    complexType = (IEdmComplexType)type.Definition;
+                }
+            }
+
+            return complexType;
+        }

```

Comment:
I think it would be easier to follow this function if you returned the complex type immediately in each branch, and returned null at the end.

```c#
if (type.IsComplex())
{
  return (IEdmComplexType) type.Definition;
}
if (type.IsCollection())
{
   // ...
   if (type.IsComplex())
   {
      return ...
    }
}
return null
```
That way the reader doesn't have to keep each branch path in the mind after it's return value is already known. It's also easier to tell that null is returned if the type is not a complex type.